pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -92,7 +92,7 @@ func (d *Deployer) TrackContainerFromBuild(build graph.Artifact, id string) {
 
 // Deploy deploys built artifacts by creating containers in the local docker daemon
 // from each artifact's image.
-func (d *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Artifact) error {
+func (d *Deployer) Deploy(ctx context.Context, out io.Writer, log *logrus.Logger, builds []graph.Artifact) error {
```

Comment:
I think we should maintain a static `logrus.Logger` reference instead of passing it around.

Reply:
https://github.com/sirupsen/logrus/issues/391 seems to suggest that multiple `*logrus.Logger` instances aren't concurrency-safe.

Reply:
you're right, I think I need to keep searching for a better solution to this problem. I'm going to close this for now